website/content/v1.10/talos-guides/configuration/insecure.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "The insecure flag"
+description: "Learn how to use the insecure flag."
+---
+
+The `--insecure` flag is a per-command argument that allows you to communicate with the Talos API when a node is in maintenance mode, that is, before a node has been configured with a machine configuration or TLS certificates.
+
+This is necessary because the machine configuration contains all the security credentials and PKI setup that secure communication between client and node.
+Without these credentials, nodes in this maintenance state can only accept API calls through the insecure interface.
+
+Only a small subset of Talos API commands support the `--insecure` flag, specifically those required for initial setup and maintenance operations.
+However, once you've applied a machine config, you must stop using the `--insecure` flag for all subsequent operations.
+The node will now expect secure communication through a talosconfig file.
+
+**Note**: The `--insecure` flag is used in a different context by the `talosctl image cache-create` command.
+This command is not used for interacting with the Talos node, but for allowing access to insecure image registries that do not support TLS.
+
+## Supported Commands With the insecure Flag
+
+The following commands can be used with the `--insecure` flag:
+
+`talosctl apply-config`
+
+Use this command alongside the `--insecure` flag to apply a machine configuration for the first time.
+
+`talosctl version`
+
+Check the Talos version running on the node.
+
+`talosctl get`
+
+Retrieves resources from the node.
+
+**Note**: You **cannot** retrieve the following resources in insecure mode:
```

Comment:
I think this list will be always outdated, so it doesn't make any sense to include it in the docs.

`talosctl get rd` gives a list of resource definitions, those which have `sensitive: true` are not available

Reply:
Is there a way to filter based on that property? This is a common question we get in slack and it would be helpful if we could do something like
```
talosctl get rd --insecure
```
to see resources that are available in insecure mode. Otherwise we could write a quick for loop that gets them.

Reply:
You can do `talosctl get rd --insecure`, but it means getting resource definitions in insecure mode. You can do `-o yaml` and filter with `yq`, but I don't think anybody needs this for real in the docs. It will be outdated anyways. If they hit a resource which is not available, it's not available, and the message is clear about it.